tests/Serializer/XmlSerializationTest.php

```diff
@@ -249,6 +250,13 @@ public function testArrayKeyValues()
         $this->assertEquals($this->getContent('array_key_values'), $this->serializer->serialize(new ObjectWithXmlKeyValuePairs(), 'xml'));
     }
 
+    public function testDeserializeArrayKeyValues()
+    {
+        $xml = $this->getContent('array_key_values_with_same_value_types');
+        $result = $this->serializer->deserialize($xml, ObjectWithXmlKeyValuePairsWithSameValueTypes::class, 'xml');
+        $this->assertEquals(new ObjectWithXmlKeyValuePairsWithSameValueTypes(), $result);
```

Comment:
can you put in the xml different values than the defaults for the property `$list` and assert that those values are set?
the thing is that this test will be true even if the deserializer simply ignores the `$list`  property